File: src/main/java/org/example/kalkulationsprogramm/service/PriceUpdateResult.java
package org.example.kalkulationsprogramm.service;

import java.util.List;

/**
 * Result object for price update operations.
 */
public record PriceUpdateResult(
        List<OfferItem> updated,
        List<OfferItem> skipped,
        List<OfferItem> unmatched
) {
}

File: src/main/kotlin/org/example/kalkulationsprogramm/service/PriceUpdateResult.kt
package org.example.kalkulationsprogramm.service
/**
 * Result object for price update operations.
 */
data class PriceUpdateResult(
    val updated: List<OfferItem>,
    val skipped: List<OfferItem>,
    val unmatched: List<OfferItem>
) {
    fun updated(): List<OfferItem> = updated
    fun skipped(): List<OfferItem> = skipped
    fun unmatched(): List<OfferItem> = unmatched
}
